cloud/microsoft/office365/exchange/mode/emailactivity.pm

```diff
@@ -58,7 +62,7 @@ sub custom_active_threshold {
 sub custom_active_output {
     my ($self, %options) = @_;
 
-    my $msg = sprintf("Active mailboxes on %s : %d/%d (%.2f%%)",
+    my $msg = sprintf("Active users on %s : %d/%d (%.2f%%)",
```

Comment:
Look at the API calls used :

`emailactivity` mode : `getEmailActivityUserDetail` [doc](https://docs.microsoft.com/en-us/graph/api/reportroot-getemailactivityuserdetail?view=graph-rest-1.0) states : _Get details about email activity **users** have performed. Specifies the date for which you would like to view the **users who performed** any activity._
`mailboxusage` mode : `getMailboxUsageDetail` [doc](https://docs.microsoft.com/en-us/graph/api/reportroot-getmailboxusagedetail?view=graph-rest-1.0) states : _Get details about **mailbox** usage._

So the `emailactivity` is about users, the `mailboxusage` is about mailboxes.
I wonder whether or not users could send mails from more than one mailbox. Think for example about shared mailboxes, for example admin@mycorp.xyz, where every admin answers to received mails with his own name (we then see "_sent by john@mycorp.xyz_" in the received mail).
I then assume this can be the case. This may be why `getEmailActivityUserDetail` API call is about users, not about mailboxes.

In addition to this, I think it's more proper / relevant to keep the unit as stated in the API documentation. Instead of making asumptions changing it :)

It's also more consistent with other modes : all other `*activity` modes are about users (`onedrive`, `sharepoint`, `skype`, `teams`). And the `*usage` modes are respectively about sites (`onedrive`, `sharepoint`) and devices (`skype`, `teams`).
This makes the whole thing properly / easily readable.

I think it's not a big problem to change one unit here (even more into the documented one), as part of a big rework of the plugin. Some Centreon Plugins' changes are sometimes breaking changes, even sometimes unit changes...
This answers your second point [here](https://github.com/centreon/centreon-plugins/pull/2042#pullrequestreview-457548963).

Reply:
And a real-life example :
I have an environment where `emailactivity` reports 380 **users**, and `mailboxusage` reports 90 **mailboxes**.
So total different numbers.
And, depending on the days, number of active **users** different than number of active **mailboxes**.
These are then 2 real different things.